index.js

```diff
@@ -1,19 +1,96 @@
+let orderArray = []
 
+// listen for user clicks to menu items
+document.addEventListener("click", (e) => {
+    if(e.target.dataset.add) {
+        addItem(e.target.dataset.add)
+    } else if(e.target.dataset.remove) {
+        removeItem(e.target.dataset.remove)
+    } 
+})
+
+// add item to order
+function addItem(itemId, quantityChange) {
+    const menuItem = menuArray.find(item => item.id.toString() === itemId)
+    if(menuItem) {
+        const existingItem = orderArray.find(item => item.id === menuItem.id)
+        
+        if(existingItem) {
+            existingItem.quantity++
+            existingItem.totalprice = existingItem.quantity * menuItem.price
+        } else {
+            const orderItem = {
+                id: menuItem.id,
+                name: menuItem.name,
+                quantity: 1,
+                totalprice: menuItem.price
+            }
+            orderArray.push(orderItem)
+        }
+        renderOrder()
+    }
+}
+
+// remove item from order
+function removeItem(itemId, quantityChange) {
+    const menuItem = menuArray.find(item => item.id.toString() === itemId)
+    if(menuItem) {
+        const existingItem = orderArray.find(item => item.id === menuItem.id)
+        
+        if(existingItem.quantity > 0) {
+            existingItem.quantity--
+            existingItem.totalprice = existingItem.quantity * menuItem.price
+        }
+        orderArray = orderArray.filter(item => {
+            return item.quantity > 0
+        })
+        renderOrder()
+    }
+}
+
+
+// render order to screen
+function renderOrder() {
+    const hasItems = orderArray.length > 0
+    orderSummary.style.visibility = hasItems ? "visible" : "hidden"
+
+    const orderItems = orderArray.map(item => `
+        <div class="order-item">
+            <h3 class="order-item-title">${item.name}</h3>
+            <p class="quantity">Quantity: ${item.quantity}</p>
+            <button id="remove-btn" class="remove-btn" data-remove="${item.id}">remove</button>
+            <p class="order-item-price">$ ${item.totalprice}</p>
+        </div>
+        `
+    ).join("")
+
+    const orderTotal = orderArray.reduce((total, item) => total + item.totalprice, 0)
+
+    orderedItemsContainer.innerHTML = orderItems
+    totalOrderPrice.textContent = `$ ${orderTotal}`
+    
+}
+
+// render menu items to screen
 function renderMenu() {
-    return menuArray.forEach(item => {
-        menu.innerHTML+=
-            `<div class="menu-item">
-                <p class="emoji">${item.emoji}</p>
-                <div class="menu-item-info-container">
-                    <h3 class="menu-item-title">${item.name}</h3>
-                    <p class="menu-item-ingredients">${item.ingredients}</p>
-                    <p class="price">$${item.price}</p>
-                </div>
-                <button id="add-btn" class="add-btn">+</button>
-            </div>`
-    })
+    const menuItems = menuArray.map(item => `
```

Comment:
I was saying:

let html = ""
menuArray.forEach(item => {
      html += (the dynamic HTML you want to inject)
})

menu.innerHTML = html